src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -5188,6 +5193,254 @@ private static List<CompletionResult> CompleteRequires(CompletionContext context
             ' ',
         };
 
+        private static List<CompletionResult> CompleteCommentHelp(CompletionContext context, ref int replacementIndex, ref int replacementLength)
+        {
+            // Finds comment keywords like ".DESCRIPTION"
+            MatchCollection usedKeywords = Regex.Matches(context.TokenAtCursor.Text, @"(?<=^\s*\.)\w*", RegexOptions.Multiline);
+            if (usedKeywords.Count == 0)
+            {
+                return null;
+            }
+
+            // Last keyword at or before the cursor
+            Match lineKeyword = null;
+            foreach (Match keyword in usedKeywords)
+            {
+                if (context.CursorPosition.Offset >= keyword.Index + context.TokenAtCursor.Extent.StartOffset)
+                {
+                    lineKeyword = keyword;
+                }
+            }
+
+            if (lineKeyword is null)
+            {
+                return null;
+            }
+
+            // Cursor is within or at the start/end of the keyword
+            if (context.CursorPosition.Offset <= lineKeyword.Index + lineKeyword.Length + context.TokenAtCursor.Extent.StartOffset)
```

Comment:
Why not do it in the previous cycle?

Reply:
Why should we? I don't think moving that entire code block into the if statement of the previous for loop makes things faster or more readable.

Reply:
I see your implementation reflects your step-by-step thinking. I have no strong doubts about performance in a completion code but for me putting one condition in one cycle and second condition in second cycle looks strange. I don't ask you to change this now until @rjmholt has doubts about the performance of a code in the PR as a whole.

Reply:
The logic here looks fine to me. It's a cheap condition, and it separates the code into a nice `if`/`return` cascade, which I think is very readable (certainly more than embedding a particular case into a loop).